handler/push.go

```diff
@@ -162,9 +163,10 @@ func Push(
 				handler,
 			)))
 
-	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		fmt.Println(r.Context())
 		mtx.Lock()
-		ps = params
+		ctx = r.Context()
```

Comment:
I think you don't need this hack anymore. It was only needed for the httprouter.Params as they wouldn't play with the promhttp instrumentation.

The whole wrapping function can probably go. Locking and context retrieval can happen up there in the `handler` function.